core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -78,6 +82,10 @@
     this.sortedDeletesByPartition = sortedDeletesByPartition;
   }
 
+  public boolean isEmpty() {
+    return (globalDeletes == null || globalDeletes.length == 0) && sortedDeletesByPartition.isEmpty();
```

Comment:
Should we have a Precondition that sortedDeletesByPartition is not null in the constructor?

Reply:
And possibly for specsByID? I'm not sure the project policy on null guarding